phoenix-core/src/main/java/org/apache/phoenix/schema/types/PDataType.java

```diff
@@ -1144,17 +1144,18 @@ public static PDataType fromLiteral(Object value) {
       return null;
     }
     for (PDataType type : PDataType.values()) {
-      if (type.isArrayType()) {
-        PhoenixArray arr = (PhoenixArray) value;
-        if ((type.getSqlType() == arr.baseType.sqlType + PDataType.ARRAY_TYPE_BASE)
-            && type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      } else {
-        if (type.getJavaClass().isInstance(value)) {
-          return type;
-        }
-      }
+			if (type.isArrayType() && type.getJavaClass().isInstance(value)) {
```

Comment:
I'm still not clear what is the root of the issue, so let's start there. If the type.isArrayType() is true, then the data type should be coercible to a PhoenixArray. Why isn't it? Is it because it's a PhoenixJson?